core/domain/user_jobs_one_off_test.py

```diff
@@ -222,6 +223,80 @@ def test_multiple_users_case(self):
         self.assertEqual(output['2'], 2)
         self.assertEqual(output['1'], 1)
 
+class LongUserBiosOneOffJobTests(test_utils.GenericTestBase):
+    """Tests for the one-off username length distribution job."""
+    USER_A_EMAIL = 'a@example.com'
+    USER_A_USERNAME = 'a'
+    USER_A_BIO = 'I am less than 500'
+    USER_B_EMAIL = 'b@example.com'
+    USER_B_USERNAME = 'b'
+    USER_B_BIO = 'Long Bio' *100
+    USER_C_EMAIL = 'c@example.com'
+    USER_C_USERNAME = 'cnone'
+    USER_C_BIO = 'Same Bio' *100
+
+    def _run_one_off_job(self):
+        """Runs the one-off MapReduce job."""
+        job_id = (
+            user_jobs_one_off.LongUserBiosOneOffJob.create_new())
+        user_jobs_one_off.LongUserBiosOneOffJob.enqueue(job_id)
+        self.assertEqual(
+            self.count_jobs_in_taskqueue(
+                queue_name=taskqueue_services.QUEUE_NAME_DEFAULT),
+            1)
+        self.process_and_flush_pending_tasks()
+
+        stringified_output = (
+            user_jobs_one_off.LongUserBiosOneOffJob.get_output(
+                job_id))
+
+        output = []
+        if stringified_output != []:
+            output = ast.literal_eval(stringified_output[0])
+            output[0] = int(output[0])
+            output[1] = [str(i) for i in output[1]]
+            return output
+
+
+    def test_null_user_bio(self):
+        """Tests the case when userbio is None."""
+        self.signup(self.USER_C_EMAIL, self.USER_C_USERNAME)
+        result = self._run_one_off_job()
+        self.assertEqual(result, None)
+
+    def test_inbound_limit(self):
+        """Tests the case where the userbio is less than 500 characters."""
+        self.signup(self.USER_A_EMAIL, self.USER_A_USERNAME)
+        user_id_a = self.get_user_id_from_email(self.USER_A_EMAIL)
+        user_services.update_user_bio(user_id_a, self.USER_A_BIO)
+        result = self._run_one_off_job()
+        self.assertEqual(result, None)
+
+    def test_outbound_limit(self):
+        """Tests the case where the userbio is more than 500 characters."""
+        self.signup(self.USER_B_EMAIL, self.USER_B_USERNAME)
+        user_id_b = self.get_user_id_from_email(self.USER_B_EMAIL)
+        user_services.update_user_bio(user_id_b, self.USER_B_BIO)
+        result = self._run_one_off_job()
+        expected_result = [800, ['b']]
+        self.assertEqual(result, expected_result)
+
+    def test_same_userbio_limit(self):
```

Comment:
Perhaps add a test for the case where two users have two different bio lengths, too.